src/flows/pipes/Notification/endpoints/HTTP/index.ts

```diff
@@ -8,26 +8,39 @@ export default register => {
     name: 'HTTP Post',
     data: {
       auth: 'none',
+      username: '',
+      password: '',
+      token: '',
       url: 'https://www.example.com/endpoint',
     },
     component: View,
     readOnlyComponent: ReadOnly,
     generateImports: () =>
-      ['http', 'json'].map(i => `import "${i}"`).join('\n'),
+      ['http', 'influxdata/influxdb/secrets', 'json']
+        .map(i => `import "${i}"`)
+        .join('\n'),
     generateTestImports: () =>
-      ['array', 'http', 'json'].map(i => `import "${i}"`).join('\n'),
+      ['array', 'http', 'influxdata/influxdb/secrets', 'json']
+        .map(i => `import "${i}"`)
+        .join('\n'),
     generateQuery: data => {
       const headers = [['"Content-Type"', '"application/json"']]
+      let prefixSecrets = ''
 
       if (data.auth === 'basic') {
-        headers.push([
-          'Authorization',
-          `http.basicAuth(u:"${data.username}", p:"${data.password}")`,
-        ])
+        headers.push(['Authorization', 'auth'])
+        prefixSecrets = `
+          username = secrets.get(key: "${data.username}")
+          password = secrets.get(key: "${data.password}")
+          auth = http.basicAuth(u: username, p: password)
+        `
       }
 
       if (data.auth === 'bearer') {
-        headers.push(['Authorization', `"Bearer ${data.token}"`])
+        headers.push(['Authorization', `"Bearer \${token}"`])
```

Comment:
curious, why the `\` in the Bearer token?

Reply:
the token is a flux variable. created by this flux command: `token = secrets.get(key: "${data.token}")`

But flux has the same string interpolation syntax as javascript (https://docs.influxdata.com/flux/v0.x/data-types/basic/string/#interpolate-strings). So it's escaped in order to persist in the query string (for flux to use).